widgetssdk/src/test/java/com/glia/widgets/view/snackbar/SnackBarDelegateTest.kt

```diff
@@ -0,0 +1,111 @@
+
+    private val mockViewWithResources: View = mock<CoordinatorLayout> {
+        on { context } doReturn mockContext
+        on { resources } doReturn mockResources
+        on { id } doReturn viewId
+    }
+
+    private val mockDecorView: View = mock {
+        on { findViewById<View>(any()) } doReturn mockViewWithResources
+    }
+    private val mockWindow: Window = mock {
+        on { decorView } doReturn mockDecorView
+    }
+
+    private val commonActivity: Activity = mock {
+        on { window } doReturn mockWindow
+    }
+
+    private val callActivity: CallActivity = mock {
+        on { findViewById<View>(any()) } doReturn mockViewWithResources
+    }
+
+    private val chatActivity: ChatActivity = mock {
+        on { findViewById<View>(any()) } doReturn mockViewWithResources
+    }
+
+    @Test
+    fun `ChatActivitySnackBarDelegate passes the corresponding view when initialized`() {
+        ChatActivitySnackBarDelegate(chatActivity, mock(), mock())
+        verify(chatActivity).findViewById<View>(R.id.chat_view)
+    }
+
+    @Test
+    fun `CallActivitySnackBarDelegate passes the corresponding view when initialized`() {
+        CallActivitySnackBarDelegate(callActivity, mock(), mock())
+        verify(callActivity).findViewById<View>(R.id.call_view)
+    }
+
+    @Test
+    fun `CommonSnackBarDelegate passes the root view when initialized`() {
+        CommonSnackBarDelegate(commonActivity, mock(), mock())
+        verify(commonActivity, times(2)).rootView
+    }
+
+    @Test
+    fun `anchorViewId returns corresponding ids when chat or call activity is passed`() {
+        CommonSnackBarDelegate(commonActivity, mock(), mock()).apply { assertNull(anchorViewId) }
+        CallActivitySnackBarDelegate(callActivity, mock(), mock()).apply { assertEquals(R.id.buttons_layout_bg, anchorViewId) }
+        ChatActivitySnackBarDelegate(chatActivity, mock(), mock()).apply { assertEquals(R.id.chat_message_layout, anchorViewId) }
+    }
+
+    @Test
+    fun `marginBottom margin is calculated when the passed activity is not a call or chat activity`() {
+        CommonSnackBarDelegate(commonActivity, mock(), mock()).apply { assertEquals(bottomMargin, marginBottom) }
+        CallActivitySnackBarDelegate(callActivity, mock(), mock()).apply { assertNull(marginBottom) }
+        ChatActivitySnackBarDelegate(chatActivity, mock(), mock()).apply { assertNull(marginBottom) }
+    }
+
+    @Test
+    fun `background and text colors are inverse in CallActivitySnackBarDelegate`() {
+        val common = CommonSnackBarDelegate(commonActivity, mock(), mock())
+        val call = CallActivitySnackBarDelegate(callActivity, mock(), mock())
+        val chat = ChatActivitySnackBarDelegate(chatActivity, mock(), mock())
+
+        assertEquals(common.fallbackBackgroundColor, chat.fallbackBackgroundColor)
+        assertEquals(common.fallbackTextColor, chat.fallbackTextColor)
+        assertEquals(common.fallbackTextColor, call.fallbackBackgroundColor)
```

Comment:
How can `common.fallbackTextColor` be equal to two different values: `chat.fallbackTextColor` `call.fallbackBackgroundColor` at the same time? 🤔

Reply:
**Chat**, should use **Common** colors, while **Call**, should be inverted.

Reply:
In other words the text 
```
common.textcolor == chat.textcolor == call.backgroundcolor
common.backgrouncolor == chat.backgroundcolor == call.textcolor
```

Reply:
Maybe it's okay, because they are coming from different entities (chat and call). 🤔

Reply:
The values are overridden and inverted by the `CallActivitySnackBarDelegate`.